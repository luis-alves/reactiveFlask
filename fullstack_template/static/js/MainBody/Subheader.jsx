import React from "react"

class Subheader extends React.Component {
  render () {
    return (
      <h6 className='page-subheader'>This is the sub header</h6>
    )
  }
}

export default Subheader
