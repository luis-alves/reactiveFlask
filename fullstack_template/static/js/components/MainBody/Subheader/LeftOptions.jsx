import React from "react"

export default class LeftOptions extends React.Component {
  render () {
    return (
      <div className="left-options-main">
        <button><i className="icons icon-add"></i><i className="subheader-text">Add a transaction</i></button>
        <button><i className="icons icon-pencil"></i><i className="subheader-text">Edit</i></button>
        <button><i className="icons icon-download"></i><i className="subheader-text">Import</i></button>
        <button><i className="icons icon-undo"></i><i className="subheader-text">Undo</i>.</button>
        <button><i className="icons icon-redo"></i><i className="subheader-text">Redo</i></button>
      </div>
    )
  }
}
