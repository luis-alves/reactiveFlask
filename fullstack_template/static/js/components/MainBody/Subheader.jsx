import React from "react"
import LeftOptions from "./Subheader/LeftOptions"



export default class Subheader extends React.Component {
  render () {
    return (
      <div className="subheader">
        <LeftOptions />
      </div>
    )
  }
}
