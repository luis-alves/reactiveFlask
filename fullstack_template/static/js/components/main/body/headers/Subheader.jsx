import React from 'react'

import LeftOptions from './subheader/LeftOptions'


export default class Subheader extends React.Component {
  render () {
    return (
      <div className='subheader'>
        <LeftOptions />
      </div>
    )
  }
}
