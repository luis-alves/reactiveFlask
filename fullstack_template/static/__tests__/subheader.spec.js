import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import Subheader from '../js/components/main/body/headers/Subheader.jsx';

describe('Subheader component', () => {
  it('renders without exploding', () => {
    expect(
      shallow(<Subheader />).length).toEqual(1)
  })
  // it('Header has class page-header', () => {
  //   const component = shallow(<Header />);
  //   expect(component.find('.page-header').length).toEqual(1);
  // })
})

test('Check snapshot', () => {
  const wrapper = shallow(<Subheader />);
  expect(wrapper).toMatchSnapshot();
});
