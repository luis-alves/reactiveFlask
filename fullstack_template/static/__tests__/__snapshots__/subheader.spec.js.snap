// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check snapshot 1`] = `
<div
  className="subheader"
>
  <LeftOptions />
</div>
`;
